docs/changelog/3_x.rst

```diff
@@ -0,0 +1,726 @@
+    your ancestors in multiple inheritence. This used to work before 3.0:
+
+    .. code-block:: sdl
+
+        type A;
+        type B extending A;
+        type C extending A, B;
+
+    but as of 3.0, the order of ancestors must be changed to match the order of
+    the bases:
+
+    .. code-block:: sdl
+
+        type A;
+        type B extending A;
+        type C extending B, A;
+
+    This is a key instance where schemas may be incompatible with 3.0.
+
+If the upgrade-check finds any problems, fix them in your schema and squash
+your migrations.
+
+.. code-block:: bash
+
+    $ edgedb migration create --squash
+
+Then run the following command inside the project directory.
+
+.. code-block:: bash
+
+  $ edgedb project upgrade --to-testing
+
+Alternatively, specify an instance name if you aren't using a project.
+
+.. code-block:: bash
+
+  $ edgedb project upgrade --to-testing -I my_instance
+
+**Hosted instances**
+
+To upgrade a remote (hosted) instance, we recommend the following
+dump-and-restore process.
+
+1. Spin up an empty 3.0 instance. You can use one of our :ref:`deployment
+   guides <ref_guide_deployment>`, but you will need to modify some of the
+   commands to use our testing channel and the beta release.
+
+   Under Debian/Ubuntu, when adding the EdgeDB package repository, use this
+   command instead:
+
+   .. code-block:: bash
+
+       $ echo deb [signed-by=/usr/local/share/keyrings/edgedb-keyring.gpg] \
+           https://packages.edgedb.com/apt \
+           $(grep "VERSION_CODENAME=" /etc/os-release | cut -d= -f2) testing \
+           | sudo tee /etc/apt/sources.list.d/edgedb.list
+
+   Use this command for installation under Debian/Ubuntu:
+
+   .. code-block:: bash
+
+       $ sudo apt-get update && sudo apt-get install edgedb-3-beta1
+
+   Under CentOS/RHEL, use this installation command:
+
+   .. code-block:: bash
+
+       $ sudo yum install edgedb-3-beta1
+
+   In any required ``systemctl`` commands, replace ``edgedb-server-2`` with
+   ``edgedb-server-3``.
+
+   Under any Docker setups, supply the ``3.0-beta.1`` tag.
+
+2. Take your application offline, then dump your v2.x database with the CLI
+
+   .. code-block:: bash
+
+       $ edgedb dump --dsn <old dsn> --all --format dir my_database.dump/
+
+   This will dump the schema and contents of your current database to a
+   directory on your local disk called ``my_database.dump``. The directory name
+   isn't important.
+
+3. Restore the empty v3.x instance from the dump
+
+   .. code-block:: bash
+
+       $ edgedb restore --all my_database.dump/ --dsn <new dsn>
+
+   Once the restore is complete, update your application to connect to the new
+   instance.
+
+   This process will involve some downtime, specifically during steps 2 and 3.
+
+**Pre-1.0 Instances**
+
+If you're still running pre-1.0 EdgeDB instances (e.g., 1.0-beta3) and want
+to upgrade to 3.0, you would need to upgrade to version 2.x first, followed by
```

Comment:
I'd say we *recommend* you upgrade to 2.x, but it shouldn't be necessary, unless I missed some specific incompatibility.

Reply:
Hey, @fantix. Does "recommend" work here, or is this upgrade path a requirement?

Reply:
Yes sure, we only dropped the test coverage to restore pre-1.0 dumps. So it should still be available now to restore directly into 3.0 without having to go through 2.x, until some day in the future we introduce such incompatibility.